### ISSUE TITLE
Support for numpy >= 2.0

### ISSUE THREAD

**Issue by charleslparker** [stanfordmlgroup/ngboost#358]
I've done some casual testing and based on the results I'm assuming support for numpy 2.0 and above isn't present yet.  Are there plans to support it in the near future?

**Comment by ryan-wolbeck** [comment]
@charleslparker yes it will be something we support in the future. If you want to accelerate that ask I'd be happy to review a PR.

**Comment by charleslparker** [comment]
Thanks, @ryan-wolbeck .  I'm a bit swamped right now but if I have a free moment I'll take a look.  Thanks for excellent library!

**Comment by jerome-f** [comment]
+1 on this Score -> grad np.linalg.solve fails because 2.0 

Changed in version 2.0: The b array is only treated as a shape (M,) column vector if it is exactly 1-dimensional. In all other instances it is treated as a stack of (M, K) matrices. Previously b would be treated as a stack of (M,) vectors if b.ndim was equal to a.ndim - 1.

**Comment by sharedw** [comment]
@jerome-f did you ever find a workaround for this?

**Comment by jerome-f** [comment]
@sharedw not really I am working with binary classification so the Fisher Information matrix is a scalar and calculating the natural gradient reduces to grad/metric. This should hold for other scalar FIM (single parameter distributions) but I have not tested them. Since the api supports multi-parameter distribution this is not a functional fix for the api.

**Comment by jerome-f** [comment]
It would be pragmatic to fix this rather than sticking to older version. 
1.26.4 is the most stable version of numpy where ngboost doesn't encounter this issue.

**Comment by anpatton** [comment]
Colab uses 1.26.4 in their default setup which is a good indicator.

**Comment by alejandroschuler** [comment]
It can't be that hard to fix it so the package works with numpy2. There's not that much code in the whole package :)

**Comment by robertsont** [comment]
I've submitted [#365](https://github.com/stanfordmlgroup/ngboost/pull/365) which fixes the issue with numpy 2.*. @jerome-f was exactly right.

**Comment by mshearer0** [comment]
Until this is fixed, for newbies searching for a way to resolved this error:

ValueError: solve: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (m,m),(m,n)->(m,n) (size 404 is different from 2)

Revert to numpy==1.26.4

**Comment by ryan-wolbeck** [comment]
This should be fixed within the next day and published out via the PR @robertsont submitted.

**Comment by ryan-wolbeck** [comment]
This has been released in 0.5.2
